index.d.ts

```diff
@@ -0,0 +1,128 @@
+declare module 'fluture' {
+  interface ResolveFunction<T> {
+    (value: T): void
+  }
+  interface RejectFunction<S> {
+    (error: S): void
+  }
+
+  interface FunctionReturning<T> {
+    (...t: any[]): T
+  }
+  abstract class Future<T, S> {
+    done<T, S> (fn: (err: S, value: T) => void): Future<T, S>
+
+    fork (reject: RejectFunction<S>, resolve: ResolveFunction<T>): void
+  }
+
+  function after<T, S> (duration: number, value: string): Future<T, S>
+
+  function and<A, B> (a: Future<A, any>, b: Future<B, any>): Future<A | B, any>
+
+  // TODO: Need Help
+  function ap<T, S> (): Future<T, S>
+
+  function attempt<T, S> (f: () => T): Future<T, S>
```

Comment:
Is it possible in TypeScript to capture the type of the values that may be thrown from a function, similarly to how Java has `void fn() throws TypeA, TypeB {...}`?